### ISSUE TITLE
Missing documentation

### ISSUE THREAD

**Issue by lluiscab** [wvffle/goap.js#1]
It's possible to kind of understand how this library is suposed to be used by looking at [planSpec.js](https://github.com/wvffle/goap/blob/master/spec/planSpec.js) but I'd like to see a proper documentation explaining how everything is tied up together and maybe some examples.

**Comment by wvffle** [comment]
Yup, I was planning to work on it, but currently I have no time to do that ;/
